Sources/PointFree/Stripe.swift

```diff
@@ -0,0 +1,220 @@
+      case year
+    }
+
+    public enum Currency: String, Codable {
+      case usd
+    }
+
+    public enum Id: String, Codable, RawRepresentable {
+      case yearly
+      case monthly
+      case yearlyTeam = "yearly-team"
+      case monthlyTeam = "monthly-team"
+
+      static let all: [Id] = [.yearly, .monthly, .yearlyTeam, .monthlyTeam]
+    }
+  }
+
+  public struct PlansEnvelope: Codable {
+    let data: [Plan]
+    let hasMore: Bool
+
+    private enum CodingKeys: String, CodingKey {
+      case data
+      case hasMore = "has_more"
+    }
+  }
+
+  public struct Token: Codable {
+    let id: String
+  }
+}
+
+//private let subscriptionPlans = fetchPlans
+//  .map(^\.data >>> filter(StripeSubscriptionPlan.Id.all.contains <<< ^\.id))
+
+private func cancelSubscription(id: String) -> EitherIO<Prelude.Unit, Stripe.Subscription> {
+  return stripeDataTask("https://api.stripe.com/v1/subscriptions/\(id)", .delete)
+}
+
+private func createCustomer(token: String) -> EitherIO<Prelude.Unit, Stripe.Customer> {
+  return stripeDataTask("https://api.stripe.com/v1/customers", .post(["source": token]))
+}
+
+private func createSubscription(customer: String, plan: Stripe.Plan.Id)
+  -> EitherIO<Prelude.Unit, Stripe.Subscription> {
+
+    return stripeDataTask("https://api.stripe.com/v1/subscriptions", .post([
+      "customer": customer,
+      "items[0][plan]": plan.rawValue
+      ]))
+}
+
+private func fetchCustomer(id: String) -> EitherIO<Prelude.Unit, Stripe.Customer> {
+  return stripeDataTask("https://api.stripe.com/v1/customers/\(id)")
+}
+
+private let fetchPlans: EitherIO<Prelude.Unit, Stripe.PlansEnvelope> =
+  stripeDataTask("https://api.stripe.com/v1/plans")
+
+private func fetchPlan(id: Stripe.Plan.Id) -> EitherIO<Prelude.Unit, Stripe.Plan> {
+  return stripeDataTask("https://api.stripe.com/v1/plans/\(id.rawValue)")
+}
+
+private func fetchSubscription(id: String) -> EitherIO<Prelude.Unit, Stripe.Subscription> {
+  return stripeDataTask("https://api.stripe.com/v1/subscriptions/\(id)")
+}
+
+private let stripeJsonDecoder: JSONDecoder = {
+  let decoder = JSONDecoder()
+  decoder.dateDecodingStrategy = .secondsSince1970
+  return decoder
+}()
+
+private enum Method {
+  case get
+  case post([String: String])
+  case delete
+}
+
+private func stripeDataTask<A>(_ urlString: String, _ method: Method = .get)
+  -> EitherIO<Prelude.Unit, A>
+  where A: Decodable {
+
+    var request = URLRequest(url: URL(string: urlString)!)
+
+    switch method {
+    case .get:
+      request.httpMethod = "GET"
+    case let .post(params):
+      request.httpMethod = "POST"
+      request.httpBody = Data(urlFormEncode(value: params).utf8)
+    case .delete:
+      request.httpMethod = "DELETE"
+    }
+
+    return pure(request)
+      .flatMap { jsonDataTask(with: auth <| $0) }
+      .withExcept(const(unit))
+}
+
+private func auth(_ request: URLRequest) -> URLRequest {
+  return request |> \.allHTTPHeaderFields %~ attachStripeAuthorization
```

Comment:
couldn't this be `let auth = \.allHTTPHeaderFields %~ attachStripeAuthorization` :D

Reply:
oh also doesnt seem this is actually used...